### PR TITLE
fix parsing errors for c#

### PR DESCRIPTION
**Pull request by cmpsr2000**
Fixes issues parsing nesting in various places, as well as folder creation.

### CODE REVIEW

File: src/Parser.py [hunk 1]
```diff
@@ -109,13 +103,25 @@ def transpile(self):
 	def class_body(self):
 		if self.expect('pass'): return
 		static = self.expect('static')
-		if self.expect('class'): self.nested_class()
+		if self.expect('@', 'tool'): self.is_tool = True
+		elif self.expect('extends'): self.base_class = self.consume()
+		elif self.expect('class_name'): self.class_name = self.consume()
```
Comment (Lcbx) [inline src/Parser.py:108]: these are handled differently between the script-level class and nested classes,
and your code currently overwrites script-level data with the last nested class info

File: src/Parser.py [hunk 2]
```diff
@@ -974,16 +993,20 @@ def consumeUntil(self, token, separator = ''):
 			result.append(self.consume())
 		return separator.join(result)
 	
-	# parse params definition (call, signal, lambda)
+	# parse params definition (call, signal, lambda) AND ENUMS!!!!
```
Comment (Lcbx) [inline src/Parser.py:996]: yeah this is unintuitive, I added a comment in main branch
sry

**Comment by Lcbx** [comment]
thanks for the PR, stranger !
I rewrote your fixes and pushed to main. feel free to add issues if you find more :)

**Comment by cmpsr2000** [comment]
@Lcbx Hello! Thanks for the review, I didn't mean to open this against the main repo but I guess my mistake worked out! I've pulled your latest main into the fork, and it still produces some incorrect output. I'll open some issues for the specific problems with simple repro cases. :+1:

**Pull request closed without merging**